src/neural/blas/winograd_convolution3.h

```diff
@@ -0,0 +1,84 @@
+/*
+ This file is part of Leela Chess Zero.
+ Copyright (C) 2018 The LCZero Authors
+
+ Leela Chess is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Leela Chess is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <vector>
+
+namespace lczero {
+
+// Convolution 3x3 on a 8x8 board using the Winograd algorithm.
+//
+// Ref:
+//
+// Fast Algorithms for Convolutional Neural Networks
+// https://arxiv.org/abs/1509.09308
+//
+// https://ai.intel.com/winograd/
+// https://ai.intel.com/winograd-2/
+
+// Convolution 3x3 using the Winograd algorithm
+class WinogradConvolution3 {
+ public:
+  // Create the zero-padded U matrix
+  static std::vector<float> ZeropadU(const std::vector<float>& U,
```

Comment:
Why uppercase U?

Reply:
Same answer.

Reply:
It's very common, through algebra software.
Altenatively we can use matrixU_, but U_ is fine.